### PR TITLE
Add trait method `Transaction::is_dynamic_fee`

### PR DESCRIPTION
**Pull request by emhane**
## Motivation

Migrate reth utility method https://github.com/paradigmxyz/reth/pull/12437#discussion_r1837882648

## Solution

Adds trait method `alloy_consensus::Transaction::is_dynamic_fee`

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/network/src/any/unknowns.rs [hunk 1]
```diff
@@ -146,6 +146,11 @@ impl alloy_consensus::Transaction for UnknownTypedTransaction {
         self.gas_price().or(self.max_priority_fee_per_gas()).unwrap_or_default()
     }
 
+    fn is_dynamic_fee(&self) -> bool {
+        self.fields.get_deserialized::<U128>("maxFeePerGas").is_some()
+            || self.fields.get_deserialized::<U128>("maxFeePerBlobGas").is_some()
```
Comment (mattsse) [inline crates/network/src/any/unknowns.rs:151]: can we use the `self.ty` here?

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit

File: crates/network/src/any/unknowns.rs (hunk 1 above)
Comment (emhane) [reply crates/network/src/any/unknowns.rs:151]: how? I thought ab it too, but it would be incorrect for eg deposit tx to do smthg like `self.ty() > 1`

Comment (mattsse) [reply crates/network/src/any/unknowns.rs:151]: oh, yeah we'd only catch known types, which defeats the purpose here

**Review by mattsse — APPROVED** [review]

**Pull request merged**